zokrates_core/src/proof_system/bn128/g16.rs

```diff
@@ -222,35 +220,47 @@ impl ProofSystem for G16 {
     }
 }
 
-fn serialize_vk(vk: VerifyingKey<Bn256>, dev_mode: bool) -> String {
-    let mut svk = format!(
-        "vk.alpha = {}
-vk.beta = {}
-vk.gamma = {}
-vk.delta = {}
-vk.gamma_abc.len() = {}
-{}",
-        parse_g1_hex(&vk.alpha_g1),
-        parse_g2_hex(&vk.beta_g2),
-        parse_g2_hex(&vk.gamma_g2),
-        parse_g2_hex(&vk.delta_g2),
-        vk.ic.len(),
-        vk.ic
-            .iter()
-            .enumerate()
-            .map(|(i, x)| format!("vk.gamma_abc[{}] = {}", i, parse_g1_hex(x)))
-            .collect::<Vec<_>>()
-            .join("\n")
-    );
+fn serialize_vk(vk: VerifyingKey<Bn256>, include_raw: bool) -> String {
+    let mut writer = csv::WriterBuilder::new()
+        .delimiter(b'=')
+        .from_writer(vec![]);
+
+    writer
+        .write_record(vec!["vk.alpha", parse_g1_hex(&vk.alpha_g1).as_str()])
```

Comment:
detail, but is there a way out of creating a vec here?

Reply:
yes we can use array here